js/skin.js

```diff
@@ -85,30 +86,36 @@ const Skin = createReactClass({
   },
 
   updatePlayhead: function(newPlayhead, newDuration, newBuffered, adPlayhead) {
-    this.setState({
-      currentPlayhead: Utils.ensureNumber(newPlayhead, this.state.currentPlayhead),
-      duration: Utils.ensureNumber(newDuration, this.state.duration),
-      buffered: Utils.ensureNumber(newBuffered, this.state.buffered),
-      currentAdPlayhead: Utils.ensureNumber(adPlayhead, this.state.currentAdPlayhead)
+    let promise = new Promise((resolve, reject) => {
+      this.setState(function(prevState) {
+        const duration = Utils.ensureNumber(newDuration, prevState.duration);
+        const totalTime = this.getTotalTime(duration);
+        const currentPlayhead = Utils.ensureNumber(newPlayhead, prevState.currentPlayhead);
+        const buffered = Utils.ensureNumber(newBuffered, prevState.buffered);
+        const currentAdPlayhead = Utils.ensureNumber(adPlayhead, prevState.currentAdPlayhead);
+        return {
+          currentPlayhead: currentPlayhead,
+          duration: duration,
+          buffered: buffered,
+          currentAdPlayhead: currentAdPlayhead,
+          totalTime: totalTime
+        }
+      });
+      resolve(true);
```

Comment:
Should we use `resolve` as a callback in setState function ? If the reject is possible in this `updatePlayhead` function, should we use try-catch ?

Reply:
Oh. yes! Thank you, @ALexanderSPB ! Good catch!